cpp/test/sg/genetic/evolution_test.cu

```diff
@@ -0,0 +1,347 @@
+    -1.7709871, -2.2943379, 0.0671570,  -2.2462875, -0.8166682, -1.3488045, -2.3724372, -0.6542480,
+    -1.6837887, 0.1718501,  -0.4232655, -1.9293420, -1.5524519, -0.8903348, -0.8235148, -0.7555137,
+    -1.2672423, -0.5341824, -0.0800176, -1.8341924, -2.0388451, -1.6274120, -1.0832978, -0.6836474,
+    -0.7428981, -0.6488642, -2.2992384, -0.3173651, -0.6495681, 0.0820371,  -0.2221419, -0.2825119,
+    -0.4779604, -0.5677801, -0.5407600, 0.1339569,  -0.8549058, -0.7177885, -0.4706391, -2.0992089,
+    -1.7748856, -0.8790807, -0.3359026, -1.0437502, -0.7428065, -0.5449560, 0.2120406,  -0.8962944,
+    -2.9057635, -1.8338823, -0.9476171, 0.0537955,  -0.7746540, -0.6021839, -0.9673201, -0.7290961,
+    -0.7500160, -2.1319913, -1.6356984, -2.4347284, -0.4906021, -0.1930180, -0.7118280, -0.6601136,
+    0.1714188,  -0.4826550};
+
+  std::vector<float> h_test = {
+    0.6506153,  -0.2861214, -0.4207479, -0.0879224, 0.6963105,  0.7591472,  -0.9145728, 0.3606104,
+    0.5918564,  -0.5548665, -0.4487113, 0.0824032,  0.4425484,  -0.9139633, -0.7823172, 0.0768981,
+    0.0922035,  -0.0138858, 0.9646097,  0.2624208,  -0.7190498, -0.6117298, -0.8807327, 0.2868101,
+    -0.8899322, 0.9853774,  -0.5898669, 0.6281458,  0.5219784,  -0.5437135, -0.2806136, -0.0927834,
+    -0.2291698, 0.0450774,  0.4253027,  0.6545525,  0.7031374,  -0.3601150, 0.0715214,  -0.9844534,
+    -0.8571354, -0.8157709, -0.6361769, -0.5510336, 0.4286138,  0.8863587,  -0.7481151, -0.6144726,
+    -0.7920206, -0.2917536, -0.6506116, -0.4862449, -0.0866336, -0.7439836, 0.3753550,  0.2632956,
+    -0.2270555, 0.1109649,  -0.6320683, 0.0280535,  0.6881603,  0.8163167,  0.1781434,  -0.8063828,
+    0.8032009,  -0.6779581, -0.8654890, -0.5322430, 0.3786414,  0.0546245,  -0.5542659, 0.6897840,
+    -0.1039676, -0.0343101, 0.4219748,  -0.4535081, 0.7228620,  0.3873561,  0.1427819,  -0.2881901,
+    0.5431166,  -0.0090170, -0.8354108, -0.0099369, -0.5904349, 0.2928394,  0.3634137,  -0.7485119,
+    -0.5442900, 0.4072478,  -0.4909732, 0.0737537,  -0.0973075, -0.0848911, 0.7041450,  0.3288523,
+    -0.5264588, -0.5135713, 0.5130192,  -0.0708379};
+
+  std::vector<float> h_testlab = {
+    -1.6506068, -1.6408135, -0.9171102, -2.2897648, -0.2806881, -0.2297245, -0.4421663, -0.7713085,
+    -1.6812845, -0.6648566, -0.5840624, -0.8432659, -0.6577426, -1.6213072, -0.2299105, -2.1316719,
+    -2.6060586, -1.8153329, 0.1657440,  -0.8794947, -1.3444440, -0.4118046, -0.3390867, -0.9532273,
+    0.0358915,  -0.6882091, -0.4517245, -0.3681215, -0.6051433, -1.0756192, -0.6731151, -1.0004896,
+    -2.4808031, -1.0080036, -1.7581659, -0.3644765, -0.2742536, -2.1790992, -1.8354263, 0.2105456,
+    -0.9973469, -0.2662037, -0.7020552, -0.7884595, -0.6079654, 0.0063403,  -1.2439414, -1.3997503,
+    -0.1228729, -0.9907357
+
+  };
+
+  std::vector<float> h_trainwts;
+  std::vector<float> h_testwts;
+
+  std::unique_ptr<rmm::device_uvector<float>> d_train, d_trainlab, d_test, d_testlab, d_trainwts,
```

Comment:
Wrapping it around wasn't my first choice, but since `rmm::device_uvector` does not have a default constructor, the gtests were failing with a `"default constructor cannot be referenced -- it is a deleted function"` error. Initializing it as `rmm::device_uvector<float> d_train(0,0);` and `resize()` -ing it later didn't help either.  So I found a way [here](https://github.com/rapidsai/cuml/pull/4077#discussion_r689686972) that wrapped it with `unique_ptr` that seemed to fix the build errors. Is the `rmm::device_uvector<float> d_train(0, 0);` what you meant?

Reply:
If you just read the [comment below](https://github.com/rapidsai/cuml/pull/4077#discussion_r692069286) what you have cited, you will see that even in that case it was possible to get rid of the extra unique_ptr wrappers. This is how it looks: [link](https://github.com/rapidsai/cuml/blob/42303fbf2d1191b15a67ce735f50610b977ebdea/cpp/test/prims/add_sub_dev_scalar.cu#L54-L86).  And this is how it would look for the current PR:
```diff
diff --git a/cpp/test/sg/genetic/evolution_test.cu b/cpp/test/sg/genetic/evolution_test.cu
index d131c185a..b48b0f4e2 100644
--- a/cpp/test/sg/genetic/evolution_test.cu
+++ b/cpp/test/sg/genetic/evolution_test.cu
@@ -39,7 +39,7 @@ namespace genetic {
  */
 class GeneticEvolutionTest : public ::testing::Test {
  public:
-  void SetUp() override
+  GeneticEvolutionTest() : d_train(0, stream)
   {
     ML::Logger::get().setLevel(CUML_LEVEL_INFO);
     CUDA_CHECK(cudaStreamCreate(&stream));
@@ -62,7 +62,7 @@ class GeneticEvolutionTest : public ::testing::Test {
     h_testwts.resize(n_tst_rows, 1.0f);
 
     // Initialize device memory
-    d_train    = std::make_unique<rmm::device_uvector<float>>(n_cols * n_tr_rows, stream);
+    d_train.resize(n_cols * n_tr_rows, stream);
     d_trainlab = std::make_unique<rmm::device_uvector<float>>(n_tr_rows, stream);
     d_test     = std::make_unique<rmm::device_uvector<float>>(n_cols * n_tst_rows, stream);
     d_testlab  = std::make_unique<rmm::device_uvector<float>>(n_tst_rows, stream);
@@ -70,7 +70,7 @@ class GeneticEvolutionTest : public ::testing::Test {
     d_testwts  = std::make_unique<rmm::device_uvector<float>>(n_tst_rows, stream);
 
     // Memcpy HtoD
-    CUDA_CHECK(cudaMemcpyAsync(d_train->data(),
+    CUDA_CHECK(cudaMemcpyAsync(d_train.data(),
                                h_train.data(),
                                n_cols * n_tr_rows * sizeof(float),
                                cudaMemcpyHostToDevice,
@@ -105,7 +105,7 @@ class GeneticEvolutionTest : public ::testing::Test {
   void TearDown() override { CUDA_CHECK(cudaStreamDestroy(stream)); }
 
   raft::handle_t handle;
-  cudaStream_t stream;
+  cudaStream_t stream = 0;
   param hyper_params;
 
   // Some mini-dataset constants
@@ -244,8 +244,8 @@ class GeneticEvolutionTest : public ::testing::Test {
   std::vector<float> h_trainwts;
   std::vector<float> h_testwts;
 
-  std::unique_ptr<rmm::device_uvector<float>> d_train, d_trainlab, d_test, d_testlab, d_trainwts,
-    d_testwts;
+  rmm::device_uvector<float> d_train;
+  std::unique_ptr<rmm::device_uvector<float>> d_trainlab, d_test, d_testlab, d_trainwts, d_testwts;
 };
 
 TEST_F(GeneticEvolutionTest, SymReg)
@@ -264,7 +264,7 @@ TEST_F(GeneticEvolutionTest, SymReg)
   cudaEventRecord(start, stream);
 
   symFit(handle,
-         d_train->data(),
+         d_train.data(),
          d_trainlab->data(),
          d_trainwts->data(),
          n_tr_rows,

```

Reply:
Okay, I got what you meant. By explicitly constructing `rmm::device_uvector` with size 0 and stream 0 in the initializer list of the default constructor of GTest, was able to get things going. Thanks tamas!

Reply:
ah, I had just implemented what you meant in the latest commit tamas (jinx!)